### ISSUE TITLE
UnhandledPromiseRejectionWarning and broken autocomplete

### ISSUE THREAD

**Issue by goto1134** [microsoft/pyright#1279]
Note: if you are reporting a wrong signature of a function or a class in the standard library, then the typeshed tracker is better suited for this report: https://github.com/python/typeshed/issues.

**Describe the bug**
pyright 1.1.95 in server mode does not autocomplete

**To Reproduce**
1. run lsp in server mode
2. use monaco language client
3. autocomplete action returns nulls and you get that output in stdout:
```
Pyright Server: (node:90) UnhandledPromiseRejectionWarning: Error: Debug Failure. False expression.
    at A._addToSourceFileListAndMap (/usr/local/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/program.ts:2074:9)
    at A.setFileOpened (/usr/local/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/program.ts:275:18)
    at t.BackgroundAnalysisProgram.setFileOpened (/usr/local/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/backgroundAnalysisProgram.ts:90:23)
    at u.setFileOpened (/usr/local/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/service.ts:192:41)
    at g.<anonymous> (/usr/local/lib/node_modules/pyright/dist/pyright-internal/src/languageServerBase.ts:789:39)
    at Generator.next (<anonymous>)
    at o (/usr/local/lib/node_modules/pyright/dist/pyright-internal.js:1:596777)
(Use `node --trace-warnings ...` to show where the warning was created)

Pyright Server: (node:90) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
(node:90) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

**Expected behavior**
I expect to have no warnings and get autocomplete after intent. It worked in 1.1.86.

**Screenshots or Code**

**VS Code extension or command-line**
I use it in language server mode, running with command `pyright-langserver --stdio`

**Additional context**

**Comment by erictraut** [comment]
Apparently your client is asking pyright to treat a non-Python file (i.e. a file that does not have a ".py" or ".pyi" file extension) as a Python source file. You're hitting an assertion that was recently added to the code. I'll add a hard check that rejects such files. This will eliminate the assert (and the associated crash), but you won't receive autocomplete actions for that file. Pyright is designed to handle only Python source files.

**Comment by goto1134** [comment]
actually, I pass file without any extension with the following URL: `inmemory://model/1`. I'll try another one with the correct extension and will tell you if that will help!

**Comment by erictraut** [comment]
This is related to another issue thread: https://github.com/microsoft/pylance-release/issues/739#issuecomment-745452422. Based on that discussion, we're considering changing our approach and allowing files that don't have ".py" or ".pyi" file extensions.

**Comment by jakebailey** [comment]
It's also microsoft/pylance-release#737, as VSC sends unsaved new files as `untitled://Untitled-1` or similar.

**Comment by erictraut** [comment]
This is now addressed in Pyright 1.1.96, which I just published.
